### PR TITLE
[CDAP-11895] - Adds stage level metrics to pipeline detailed view

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds Records in and out metrics for each stage in a pipeline
  - If there is a time series : Render the graph
  - If not get the aggregate values: Render the number
  - If nothing show "No Data"
- Consolidates all react components imported from react to angular in one file (`react-components.js`).
- Upgrades gulp-stylelint to latest stable version and removes all deprecated rules. (no more warnings in gulp build yay!)

JIRA: https://issues.cask.co/browse/CDAP-11895

### CODE REVIEW

File: cdap-ui/app/directives/react-components/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+angular.module(PKG.name + '.commons')
+  .directive('caskHeader', function(reactDirective) {
+    return reactDirective(window.CaskCommon.Header);
+  })
+  .directive('keyValuePairs', function(reactDirective) {
+    return reactDirective(window.CaskCommon.KeyValuePairs);
+  })
+  .directive('keyValuePairs', function(reactDirective) {
+    return reactDirective(window.CaskCommon.KeyValuePairs);
```
Comment (tonybach) [inline cdap-ui/app/directives/react-components/index.js]: duplicate?

File: cdap-ui/app/directives/react-components/index.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  })
+  .directive('dataprep', (reactDirective) => {
+    return reactDirective(window.CaskCommon.DataPrepHome);
+  })
+  .directive('caskResourceCenterButton', function(reactDirective) {
+    return reactDirective(window.CaskCommon.ResourceCenterButton);
+  })
+  .directive('caskResourceCenterButton', function(reactDirective) {
+    return reactDirective(window.CaskCommon.ResourceCenterButton);
```
Comment (tonybach) [inline cdap-ui/app/directives/react-components/index.js]: here too

File: cdap-ui/package.json [hunk 3]
```diff
@@ -169,4 +170,4 @@
       "^mocks": "<rootDir>/__mocks__"
     }
   }
-}
+}
```
Comment (tonybach) [inline cdap-ui/package.json]: Add new line

**Review by tonybach — COMMENTED** [review]
Some minor comments

**Comment by tonybach** [comment]
Code LGTM 👍

**Comment by leacd** [comment]
LGTM

**Pull request merged**